src/ReplicatedStorage/Client/InstanceComponents/Misc/Sprint.lua

```diff
@@ -0,0 +1,63 @@
+local Players = game:GetService("Players")
+local ReplicatedStorage = game:GetService("ReplicatedStorage")
+local ContextActionService = game:GetService("ContextActionService")
+local Trove = require(ReplicatedStorage.Packages.Trove)
+local Comm = require(ReplicatedStorage.Packages.Comm)
+
+local Sprint = {}
+Sprint.__index = Sprint
+
+function Sprint.new(character: Model)
+	local clientComm = Comm.ClientComm.new(character, false, "Sprint")
+	local self = setmetatable({}, Sprint)
+
+	self._character = character
+	self._comm = clientComm
+	self._trove = Trove.new()
+	self._trove:Add(self._comm)
+	self._serverObject = clientComm:BuildObject()
+
+	self._owner = Players:GetPlayerFromCharacter(character.Parent)
+	character.AncestryChanged:Connect(function()
+		self._owner = Players:GetPlayerFromCharacter(character.Parent)
+	end)
```

Comment:
This is my bad but make sure to add this to the trove. (`self._trove:Connect(character.AncestryChanged, function()...`)